test: Intermittent failure in `feature_index_prune.py`

https://cirrus-ci.com/task/5021369209978880
https://cirrus-ci.com/task/6533221721571328
https://cirrus-ci.com/task/5355957631320064

Errors are quite different:

- > https://cirrus-ci.com/task/5021369209978880 and https://cirrus-ci.com/task/6533221721571328

```
test_framework.authproxy.JSONRPCException: 'getbestblockhash' RPC took longer than 240.000000 seconds. Consider using larger timeout for calls that take longer to return. (-344)
```

- > https://cirrus-ci.com/task/5355957631320064

```
AssertionError: [node 1] Expected messages "['limited pruning to height 2489']" does not partially match log:
```

--- a/test/functional/feature_index_prune.py
+++ b/test/functional/feature_index_prune.py
@@ -138,6 +138,7 @@ class FeatureIndexPruneTest(BitcoinTestFramework):
             self.connect_nodes(i, 3)
 
         self.sync_blocks(timeout=300)
+        self.sync_index(height=2500)
 
         for node in self.nodes[:2]:
             with node.assert_debug_log(['limited pruning to height 2489']):

